File: aftarobotlibrary3/lib/data/vehicle_location.dart
import 'package:aftarobotlibrary3/data/vehicledto.dart';

class VehicleLocation {
  VehicleDTO vehicle;
  double latitude;
  double longitude;
  int timestamp;
  String date;

  VehicleLocation({
    this.vehicle,
    this.latitude,
    this.longitude,
    this.timestamp,
    this.date,
  });

  VehicleLocation.fromJson(Map data) {
    if (data['vehicle'] != null) {
      vehicle = VehicleDTO.fromJson(data['vehicle']);
    }

    this.timestamp = data['timestamp'];
    this.date = data['date'];

    if (data['latitude'] is int) {
      this.latitude = data['latitude'] * 1.0;
    } else {
      this.latitude = data['latitude'];
    }
    if (data['longitude'] is int) {
      this.longitude = data['longitude'] * 1.0;
    } else {
      this.longitude = data['longitude'];
    }
  }

  Map<String, dynamic> toJson() {
    var vh;
    if (vehicle != null) {
      vh = vehicle.toJson();
    }
    Map<String, dynamic> map = {
      'vehicle': vh,
      'latitude': latitude,
      'longitude': longitude,
      'timestamp': timestamp,
      'date': date,
    };
    return map;
  }
}

File: commuter/android/app/src/main/java/com/aftarobot/commuter/util/VehicleLocation.java
package com.aftarobot.commuter.util;

public class VehicleLocation {
    private VehicleDTO vehicle;
    private long timestamp;
    private String date;
    private double latitude, longitude;

    public VehicleLocation() {
    }

    public VehicleLocation(VehicleDTO vehicle, long timestamp, String date, double latitude, double longitude) {
        this.vehicle = vehicle;
        this.timestamp = timestamp;
        this.date = date;
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public VehicleDTO getVehicle() {
        return vehicle;
    }

    public void setVehicle(VehicleDTO vehicle) {
        this.vehicle = vehicle;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }
}
